contracts/AssetHolder.sol

```diff
@@ -102,7 +102,7 @@ contract AssetHolder {
 
     }
 
-    function claimAll(bytes32 channelId, bytes32 guaranteedChannelId, bytes calldata destinationsBytes, bytes calldata allocationBytes) external {
```

Comment:
This might slightly up gas usage, but it is only adding one address and makes it easier to use. If we want to switch this back later to conserve gas we can.

Reply:
I think this is a change in the right direction, as the current code is not following the [specification](https://github.com/statechannels/nitro-protocol/blob/3dce997e0c445cd2e1cfc509f4b23984d7d4177a/docs/nitro/outcomes.md) for what kind of structure a `LabelledAllocationOrGuarantee` should be.

1. There is redundant information, as the `guaranteedChannelId` is inside `guaranteeBytes` as well as being a function parameter.
2. I don't this function is yet performing an important check: is the guarantee actually pointing to the `guaranteedChannelId`? As is, I think I can use claim any guarantee against any channel, as long as they have both been finalized on chain.

I think we could solve 1 and 2 by removing `guaranteedChannelId` from the calldata and pulling it out of the `guarantee` (after decoding it from `guaranteeBytes`) before passing it into the check against channel storage.

Reply:
If we remove `guaranteedChannelId` from `guarantee` doesn't that violate the spec?

Reply:
Updated here 44ccec6

Reply:
👍